### DISCUSSION TITLE
How to upgrade from 2.1 to 2.5?

### DISCUSSION THREAD

**Discussion by houmie** [TinCanTech/easy-tls#246] [EasyTLS and related HELP] [answered]
Hey mate,

I hope all is well with you.

I had a chance to look into this project again. I can see a lot has changed.  I'm still using release ```2.1``` and was hoping to start using ```2.5``` going forwards.

When I try to initialise it I get this message:
```
sudo ./easytls init-tls
Easy-TLS version: 2.5
Download the file from the URL above.
Note: URL: https://raw.githubusercontent.com/TinCanTech/easy-tls/master/easytls-openssl.cnf
```  

But I have already downloaded that file and don't understand how to proceed from here.

In a nutshell we used to do this in 2.1:
```
sudo ./easytls init-tls
sudo ./easytls --openvpn=/usr/sbin/openvpn build-tls-crypt-v2-server $SERVER_CRT
sudo ./easytls --openvpn=/usr/sbin/openvpn build-tls-crypt-v2-client $SERVER_CRT client1
```

I wonder if it's still backwards compatible, once I bypassed the issue further above.

Thanks

**Comment by TinCanTech** [comment, marked as answer]
Upgrading should only pose a few problems, which I can help you with.

First, `--openvpn` has been removed, it was a developer necessary option prior to OpenVPN 2.5 release.

Second, you must have `easytls-openssl.cnf` in the `easytls` folder.

Third, you may encounter some hash issues, which we should be able to fix.

**Comment by houmie** [comment]
Hi mate,

Sorry for late reply. It all worked well after following your instructions.  I have been testing v2.5 since then without any issues with an iOS client.

**Reply by TinCanTech** [reply]
Thanks for letting me know and testing ! :+1: 

As you may have noticed, I released v2.6.0 to soon. I found a better way to manage GROUP keys. So, there will be v2.7.0 very soon.  For now, all you need is this file [easytls](https://github.com/TinCanTech/easy-tls/blob/master/easytls) and you should be good to go.
